### PR TITLE
feat: Add Saved Addresses CRUD

### PR DESCRIPTION
**Pull request by emizzle**
Adds saved address CRUD API.

### CODE REVIEW

File: services/wallet/database.go
Comment (alaibe) [inline services/wallet/database.go]: nit: remove if not needed

**Review by alaibe — APPROVED** [review]
LGTM, Once this one got merged: https://github.com/status-im/status-go/pull/2349 I will add chainID here too


Out of curiousity, how is it different than Favourite? Could we not reuse them?

File: services/wallet/database.go
Comment (emizzle) [reply services/wallet/database.go]: Removed ✂️

**Comment by emizzle** [comment]
> Out of curiousity, how is it different than Favourite? Could we not reuse them?

Great question, I have no idea. Except maybe that Saved Addresses can also be favourited? 🤷‍♂️  It's very unclear from the design at this point. Maybe @iurimatias has more info?

**Comment by emizzle** [comment]
The failing test is a waku test which seems odd considering the changeset. Is this a known failing test?

**Comment by alaibe** [comment]
@emizzle There was an issue in the test which were causing them to fail all the time, if you rebase to develop and retry them (maybe more than once) they should go green

**Comment by iurimatias** [comment]
tested in https://github.com/status-im/status-desktop/pull/3425

**Pull request merged**
